chrispinkney/cinesync
# Issue-51: Movie form autopopulation when adding to a List

## Issue

Closes #51

## Type of Change



## Description

Currently, movies need to be manually added and their details populated in a movie list. Integrating [TMDB](www.themoviedb.org) as a data source will allow us to populate movie details automatically.

When editing a movie list, the user can click to add a movie, then search for their movie and once found, they can select it. The movie will be automatically added to the movie list. 

This also changes the editability of the movie list `<Datagrid/>` component, and required removal of the `<GenreDropdown/>` in favour of a `<GenreDisplay/>`, in which the chips are displayed, without a dropdown selection menu.

Additionally, tool tip styling has been changed in `theme.ts` . The background color has been changed to black at 70% transparency.

lodash has been added as a dependency to implement search text debounce.

## Testing the PR

- run backend as usual
- `cd web` -> `npm i` to install lodash
- `npm run dev`
- Create a list and go to its page
- Click the ➕ button on the list to open the add movie dialog
- Search for and add various movies. 
  - These should be added to the list, but not saved.
  - To save, click the list save icon as usual.
- Test that the movie search details are cleared when closing the dialog.

## Checklist
